### DISCUSSION TITLE
Two  Browse Mode queries

### DISCUSSION THREAD

**Discussion by TechHorseG** [nvaccess/nvda#17620] [General]
Win 10 / NVDA portable 2019.2 / Laptop layout / Review cursor follows mouse / not trapping non-Brows keys

Hello. I am getting to grips with Browse Mode in Firefox and have two  issues.

1. when BM is enabled, this seems to disable the ability to copy text via the ctrl+C command? I keep forgetting and have to go back and right-click then select "copy". Is there any way to restore the ctrl+C ability while in Browse Mode?

2. Is there a quick way to sync the BM focus to the mouse pointer location? If I am scrolling through a webpage and see a section I want to focus on and be able to tab through links, use up / down arrow reading etc, well I can't seem to find any quick way to get the brows focus to wherever I ampointing?

Thanks.

**Comment by JulienCochuyt** [comment]
> 1. when BM is enabled, this seems to disable the ability to copy text via the ctrl+C command? I keep forgetting and have to go back and right-click then select "copy". Is there any way to restore the ctrl+C ability while in Browse Mode?

Currently, you have either to:
- First press `NVDA+F2` to let next key pass through, then hit `control+C`
- Use `control+insert` instead of `control+C` to copy, but you'd probably have to maintain `control` pressed while pressing `insert` twice quickly if you use it as NVDA key.

Issue #9833 was raised to simplify this behavior: Pending PR #9834.


> 2. Is there a quick way to sync the BM focus to the mouse pointer location? If I am scrolling through a webpage and see a section I want to focus on and be able to tab through links, use up / down arrow reading etc, well I can't seem to find any quick way to get the brows focus to wherever I ampointing?

Not that "quick":
- First move the navigator object to the mouse object: `NVDA+shift+N`
- If NVDA does not say "paragraph", move to the parent object: `NVDA+shift+upArrow`
- Then move the system caret to the review cursor: Hold the NVDA key and press `backspace` twice quickly.

The above shortcuts are for laptop keyboard layout, adapt accordingly if you're using desktop layout.

**Comment by Brian1Gaff** [comment]
I think that most of us do not use the mouse so this aspect has never been a 
top priority, also as most of the developers are blind themselves its hard 
to test it.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Comment by JulienCochuyt** [comment]
@Brian1Gaff 

> I think that most of us do not use the mouse so this aspect has never been a top priority,

Historically true, but there have been numerous fixes and enhancements lately to tend towards a better support for low vision users.
Also, a good handling of text units often (yet not always) also impacts keyboard-driven content review.

> also as most of the developers are blind themselves its hard to test it.

You are welcome to count me among the volunteer developers and testers.
I'm using magnification (mainly for contrast) + speech + braille.

**Comment by TechHorseG** [comment]
Thanks for responses. Let me be counted as another one who doesn't want ctrl C to be "eaten" :)

As for the other issue, sorry to say this but ouch :) That is quite a lot of work to do something that may need to be done quite frequently.

So let me please also request simplification. Maybe something as simple as NVDA + double-click to move browse focus to the mouse poimter.

Or would it be possible to have the brows focus follow the mouse just like the review cursor can, without any user input at all?

**Comment by JulienCochuyt** [comment]
> So let me please also request simplification. Maybe something as simple as NVDA + double-click to move browse focus to the mouse poimter.

Something like this, indeed, would be my solution of choice.
We'd need combined mouse+keyboard gestures for this kind of functionality.
@leonardder: I suppose you might have considered the question with the implementation of Babbage's Magnification VisionProvider. Would you please share any thoughts regarding the potential definition of such gestures?

> Or would it be possible to have the brows focus follow the mouse just like the review cursor can, without any user input at all?

I'm really not sure this would be performance efficient, plus dozen of nasty side effects.

**Comment by Adriani90** [comment]
Re 1. native selection mode is now available for Firefox at least, so ctrl+c should not be eaten anymore in this browser with native selection enabled. For other browsers work is still in progress from browser developer's side, and the issue #5354 is still open to continue this discussion there.
Re 2. Actually it would be very annoying to have the virtual cursor automatically follow mouse cursor. If this is implemented, this should definitely be an optional setting because using touchpads would result in unexpected movements while browsing websites etc. But I agree a more intuitive way of synchronizing them on demand should be implemented. Could you open a new issue for the synchrinizing mouse cursor with virtual cursor request?
